assets/src/admin.js

```diff
@@ -60,26 +60,86 @@ jQuery(function ($) {
+            approximate:       $form.find('input#approximate'),
+            latitude:          $form.find('input[name=latitude]'),
+            longitude:         $form.find('input[name=longitude]'),
+			group:             $form.find('input#group'),
+			group_notes:       $form.find('textarea[name=group_notes]'),
+			group_status:      $form.find('input[name="group_status"]'),
+            region:            $form.find('select[name=region]'),
+            district:          $form.find('select[name=district]'),
+            website:           $form.find('input[name=website]'),
+            email:             $form.find('input[name=email]'),
+            phone:             $form.find('input[name=phone]'),
+            contact_1_name:    $form.find('input[name=contact_1_name]'),
+            contact_1_email:   $form.find('input[name=contact_1_email]'),
+            contact_1_phone:   $form.find('input[name=contact_1_phone]'),
+            contact_2_name:    $form.find('input[name=contact_2_name]'),
+            contact_2_email:   $form.find('input[name=contact_2_email]'),
+            contact_2_phone:   $form.find('input[name=contact_2_phone]'),
+            contact_3_name:    $form.find('input[name=contact_3_name]'),
+            contact_3_email:   $form.find('input[name=contact_3_email]'),
+            contact_3_phone:   $form.find('input[name=contact_3_phone]'),
+            mailing_address:   $form.find('input[name=mailing_address]'),
+            venmo:             $form.find('input[name=venmo]'),
+            last_contact:      $form.find('input[name=last_contact]'),
+        };
+
+        // set a state for $fields 
+        //    state: loading, error, warning
+        //    code:  a code that corresponds with a field <small data-message> attribute
+        $.fn.setState = function(state, code) {
+            var $field = $(this);
+            $field.siblings('[data-message]').removeClass('show');
+            $field.removeClass('error warning');
+            if (-1 === ['warning','loading','error'].indexOf(state)) {
+                state = '';
+            }
+            $field.data('state', state);
+            if ('error' === state || 'warning' === state) {
+                $field.addClass(state);
+                if (code) {
+                    $field.siblings('[data-message=' + code + ']').addClass('show');
+                }
+            }
+            updateFormState();
+            return this;
+        };
+
+        // after field states update, toggle publish button
+        function updateFormState() {
+            var pendingFields = [];
+            Object.keys($fields).forEach(function(field) {
+                var state = $fields[field].data('state');
+                if (-1 < ['error','loading'].indexOf(state)) {
+                    pendingFields.push($fields[field][0])
+                }
+            });
+            form_valid = !pendingFields.length;
```

Comment:
later, we could use `pendingFields` as an array of DOM elements for hyperlinks, to anchor link users to in some top message. "The following fields need attention: ...."